src/NexusMods.App.UI/Pages/ItemContentsFileTree/ItemContentsFileTreeViewModel.cs

```diff
@@ -35,64 +38,106 @@ public ItemContentsFileTreeViewModel(
+            })
+            .AddTo(ref _disposables);
 
-            var behavior = workspaceController.GetOpenPageBehavior(pageData, info);
-            var workspaceId = workspaceController.ActiveWorkspaceId;
-            workspaceController.OpenPage(workspaceId, pageData, behavior);
-        },
-        this.WhenAnyValue(vm => vm.SelectedItem)
+        RemoveCommand = this.ObservePropertyChanged(vm => vm.SelectedItem)
             .WhereNotNull()
-            .Select(item => item is { IsFile: true, IsDeletion: false })
-        );
+            .Select(_ => true)
+            .ToReactiveCommand<R3.Unit>(async (_, cancellationToken) =>
+                {
+                    var gamePath = SelectedItem!.Key;
+                    var group = LoadoutItemGroup.Load(connection.Db, Context!.GroupId);
+                    var loadoutItemsToDelete = group
+                        .Children
+                        .OfTypeLoadoutItemWithTargetPath()
+                        .Where(item => item.TargetPath.Item2.Equals(gamePath.LocationId) && item.TargetPath.Item3.StartsWith(gamePath.Path))
+                        .ToArray();
+                    
+                    if (loadoutItemsToDelete.Length == 0)
+                    {
+                        logger.LogError("Unable to find Loadout files with path `{Path}` in group `{Group}`", gamePath, group.AsLoadoutItem().Name);
+                        return;
+                    }
+
+                    using var tx = connection.BeginTransaction();
+                
+                    foreach (var loadoutItem in loadoutItemsToDelete)
+                    {
+                        tx.Delete(loadoutItem, recursive: false);
+                    }
+                    
+                    await tx.Commit();
+                
+                    // Refresh the file tree, currently by re-creating it which isn't super great
+                    FileTreeViewModel = new LoadoutItemGroupFileTreeViewModel(group.Rebase());
+                }
+            )
+            .AddTo(ref _disposables);
 
         this.WhenActivated(disposables =>
         {
-            this.WhenAnyValue(vm => vm.Context)
+            // Populate the file tree
+            this.ObservePropertyChanged(vm => vm.Context)
                 .WhereNotNull()
                 .Select(context => LoadoutItemGroup.Load(connection.Db, context.GroupId))
                 .Where(group => group.IsValid())
                 .Do(group => TabTitle = group.AsLoadoutItem().Name)
                 .Select(group => new LoadoutItemGroupFileTreeViewModel(group))
-                .BindToVM(this, vm => vm.FileTreeViewModel)
+                .Do(viewModel => FileTreeViewModel = viewModel)
+                .Subscribe()
                 .DisposeWith(disposables);
-
+            
+            // Observe selected items
             this.WhenAnyValue(vm => vm.FileTreeViewModel!.TreeSource.Selection)
+                .ToObservable()
```

Comment:
@erri120 Do you know if there is a better way to do this with R3 that doesn't require using system observables? 

`ObservePropertyChanged` only supports immediate properties, no chains and no nullable chains.